commons/src/test/java/io/aiven/kafka/tieredstorage/commons/security/AesEncryptionProviderTest.java

```diff
@@ -24,22 +24,28 @@
 
 import static org.assertj.core.api.Assertions.assertThat;
 
-public class EncryptionProviderTest extends RsaKeyAwareTest {
+public class AesEncryptionProviderTest extends RsaKeyAwareTest {
 
     @Test
     void alwaysGeneratesNewKey() {
         final AesEncryptionProvider aesProvider = new AesEncryptionProvider();
-        final var dataKey1 = aesProvider.createDataKey();
-        final var dataKey2 = aesProvider.createDataKey();
+        final DataKeyAndAAD dataKeyAndAad1 = aesProvider.createDataKeyAndAAD();
+        final DataKeyAndAAD dataKeyAndAad2 = aesProvider.createDataKeyAndAAD();
 
-        assertThat(dataKey1).isNotEqualTo(dataKey2);
+        assertThat(dataKeyAndAad1).isNotEqualTo(dataKeyAndAad2);
+    }
+
+    @Test
+    void keyMustBeDifferentFromAAD() {
+        final DataKeyAndAAD dataKeyAndAad = new AesEncryptionProvider().createDataKeyAndAAD();
+        assertThat(dataKeyAndAad.dataKey.getEncoded()).isNotEqualTo(dataKeyAndAad.aad);
```

Comment:
Should we verify that AAD is actually there?

Reply:
Yes, added the `keyAndAADMustBePresent` test.